### DISCUSSION TITLE
OpenShell on Surface ( ARM x86 )

### DISCUSSION THREAD

**Discussion by ascdew** [Open-Shell/Open-Shell-Menu#949] [General]
Has anyone had any success running openshell on the Surface?  We are doing some initial test rollouts, sadly we could only get the Surface with the ARM x86 chip set.   We did update Win10 from Home to Pro - however while the openshell installer ran without issue - openshell will not run.

Perhaps we need to recompile the source for the ARM chipset, but wanted to query the group to see if anyone else has had success or failure with openshell on the Surface before we go down that rabbit hole?   Also we tested with V4.4.160.

Would appreciate any comments/suggestions!

**Comment by SirBlain** [comment]
I have been working on this for a few days. I have the ARM SQ1 Surface Pro X. 
Same thing, installer runs, Open-Shell gives me a black screen upon startup. 

If there is anything I can do or any info I can gather I would be happy to provide it.
I tested with V4.4.160, and V4.4.170, I am on Windows 11, build 22621.

**Comment by blackcrack** [comment]
Arm7 & Arm v8(64Bit) is a total other Architecture and have nothing 
to do with a x86/x86_64 IBM-Compatible Achitecture,
you can't compare Pear with Apples... so, 
Open shell is a Programm for a Desktop Computer like them from AMD and Intel, so it is need Windows 7 up to windows 10,
Windows 11 is it not supported, unless you get a programmer that inserts the required cod parts for Win11.
best

**Comment by Ibuprophen** [comment]
@ascdew & @SirBlain, This software is not compatible to run on an Windows ARM/ARM64 based OS (even if it's an Emulation Based OS). 

As @blackcrack stated, This is primarily because of the Architecture Compatibility. Your running (what's called) an AArch Processor (aka ARM or ARM32). 

There would require ALLOT of work for this software to be restructured for this type of UWP (Universal Windows Platform) Compatibility. 

There's much more to this as I had done my best to explain the above in a simplified way. 

Thank you very much for your time, support and understanding with this software and I hate to tell you that it can't be used on your device. 

~Ibuprophen

**Comment by ascdew** [comment]
I figured out as much in looking thru the base code, but appreciate the
response / confirmation.

Thanks!

**Comment by SirBlain** [comment]
@blackcrack and @Ibuprophen appreciate the clarification and response, thank you. Running great on my Intel machine and I had a feeling it was an Arm compatibility problem. Also thanks for all the hard work.

**Comment by marypcbuk** [comment]
> Perhaps we need to recompile the source for the ARM chipset,

this would be the first step, although Windows on Arm can run x86 apps in emulation and even X64 apps in Windows 11, but while the current build installs, I can't actually get the UI to open. There are likely to be different code paths for various things in the internals of Windows on Arm. It's not the absolute impossibility suggested ahead and you're certainly not limited to the UWP apps that ran on Windows RT, but it would be a good deal of work. Disappointingly - because Windows on Arm users are being pushed to Widows 11 for features like 64-bit emulation - I am still hunting for any tools to improve the Windows 11 UI on Arm.
